lib/appsignal/hooks/puma.rb

```diff
@@ -33,5 +37,57 @@ def stop_workers
         end
       end
     end
+
+    class PumaProbe
+      def initialize
+        @hostname = Appsignal.config[:hostname] || Socket.gethostname
+      end
+
+      def call
+        return unless ::Puma.stats
+
+        stats = JSON.parse Puma.stats, :symbolize_names => true
+        counts = {
+          :backlog => 0,
+          :running => 0,
+          :pool_capacity => 0,
+          :max_threads => 0
+        }
+
+        if stats[:worker_status] # Multiple workers
+          stats[:worker_status].each do |worker|
+            stat = worker[:last_status]
+
+            counts[:backlog] += stat[:backlog]
+            counts[:running] += stat[:running]
+            counts[:pool_capacity] += stat[:pool_capacity]
+            counts[:max_threads] += stat[:max_threads]
+          end
+
+          gauge(:workers, stats[:workers], :kind => :count)
+          gauge(:workers, stats[:booted_workers], :kind => :booted)
+          gauge(:workers, stats[:old_workers], :kind => :old)
```

Comment:
Since these only show up in clustered mode, do we still add these metrics to the default Puma dashboard?
Would people run it in single worker mode a lot?

Reply:
According to @jvanbaarsen they don't run in single worker mode a lot.

Reply:
ok! Then no need to split it in two detected metric dashboards :)